src/js/media-captions-menu-button.js

```diff
@@ -0,0 +1,184 @@
+</svg>`;
+
+const ccDisabledIcon = `
+<svg class="disabled" slot="button-content" aria-hidden="true" viewBox="0 0 26 24">
+  <path d="M17.73 14.09a1.4 1.4 0 0 1-1 .37 1.579 1.579 0 0 1-1.27-.58A3 3 0 0 1 15 12a2.8 2.8 0 0 1 .5-1.85 1.63 1.63 0 0 1 1.29-.57 1.47 1.47 0 0 1 1.51 1.2l1.43-.34A2.89 2.89 0 0 0 19 9.07a3 3 0 0 0-2.14-.78 3.14 3.14 0 0 0-2.42 1 3.91 3.91 0 0 0-.93 2.78 3.74 3.74 0 0 0 .92 2.66 3.07 3.07 0 0 0 2.34 1 3.07 3.07 0 0 0 1.91-.57 3.17 3.17 0 0 0 1.07-1.74l-1.4-.45c-.083.43-.3.822-.62 1.12Zm-7.22 0a1.43 1.43 0 0 1-1 .37 1.58 1.58 0 0 1-1.27-.58A3 3 0 0 1 7.76 12a2.8 2.8 0 0 1 .5-1.85 1.63 1.63 0 0 1 1.29-.57 1.47 1.47 0 0 1 1.51 1.2l1.43-.34a2.81 2.81 0 0 0-.74-1.32 2.94 2.94 0 0 0-2.13-.78 3.18 3.18 0 0 0-2.43 1 4 4 0 0 0-.92 2.78 3.74 3.74 0 0 0 .92 2.66 3.07 3.07 0 0 0 2.34 1 3.07 3.07 0 0 0 1.91-.57 3.23 3.23 0 0 0 1.07-1.74l-1.4-.45a2.06 2.06 0 0 1-.6 1.07Zm12.32-8.41a2.59 2.59 0 0 0-2.3-2.51C18.72 3.05 15.86 3 13 3c-2.86 0-5.72.05-7.53.17a2.59 2.59 0 0 0-2.3 2.51c-.23 4.207-.23 8.423 0 12.63a2.57 2.57 0 0 0 2.3 2.5c1.81.13 4.67.19 7.53.19 2.86 0 5.72-.06 7.53-.19a2.57 2.57 0 0 0 2.3-2.5c.23-4.207.23-8.423 0-12.63Zm-1.49 12.53a1.11 1.11 0 0 1-.91 1.11c-1.67.11-4.45.18-7.43.18-2.98 0-5.76-.07-7.43-.18a1.11 1.11 0 0 1-.91-1.11c-.21-4.14-.21-8.29 0-12.43a1.11 1.11 0 0 1 .91-1.11C7.24 4.56 10 4.49 13 4.49s5.76.07 7.43.18a1.11 1.11 0 0 1 .91 1.11c.21 4.14.21 8.29 0 12.43Z"/>
+</svg>`;
+
+const template = document.createElement('template');
+
+template.innerHTML = `
+<style>
+  :host,
+  media-chrome-menu-button {
+    display: contents;
+  }
+
+  media-captions-listbox {
+    position: absolute;
+    bottom: 44px;
+    width: 100px;
+    max-height: 300px;
+    overflow: auto;
+  }
+
+  /*
+   * if no captions selected, enabled icon should be hidden
+   * if captions selected, the disabled icon should be hidden
+   */
+  media-chrome-menu-button:not(.enabled) .enabled,
+  media-chrome-menu-button.enabled .disabled {
+    display: none;
+  }
+  /*
+   * if captions selected, enabled icon should be shown
+   * if no captions selected, disabled icon should be shown
+   */
+  media-chrome-menu-button.enabled .enabled {
+  media-chrome-menu-button:not(.enabled) .disabled {
+    display: initial;
+  }
+</style>
+
+<media-chrome-menu-button aria-label="captions menu button">
+  ${ccEnabledIcon}
+  ${ccDisabledIcon}
+  <media-captions-listbox slot="listbox" media-controller="mc"></media-captions-listbox>
```

Comment:
should this have a `media-controller` attr?

Reply:
Good catch, I thought I addressed it already but I forgot.

It should get the captions-menu-button's `media-controller` if it's given, but should work otherwise. Definitely needs to get updated before this PR can be merged.